system-tests/README.md

```diff
@@ -19,3 +19,35 @@ It uses Docker containers to create containerised instances of Kafka and other c
 * Wait
 
 Note: these tests take some time to run.
+
+
+### General Architecture
+
+
+The system tests use pytest for the test runner, and use separate fixtures for different configurations of the file-writer. 
+
+Firstly, the system tests attempt to build and tag the latest file-writer image. This can take a lot of time especially if something in conan has changed, as it has to reinstall all of the conan packages.
```

Comment:
Should mention this is a layer-cached build. Note that there are three main layers, the base image, all system and conan dependencies installed, and the file writer itself. This strikes a balance between time taken to rebuild the image when something changes, and the disk space used by the cached layers.